### PR TITLE
feat(basemaps): To be able to ready basemaps config json file that been gunzipped.

### PR DESCRIPTION
**Pull request by Wentao-Kuang**
#### Motivation
The argo-tasks using latest version of `@chunkd/fs` which removed the logic to ready gz json files. However, create-mapsheet cli require to ready gz config file from s3.

#### Modification
Add the logic of reading gz file into JSON object.

#### Checklist

- [ ] Tests updated
- [ ] Docs updated
- [ ] Issue linked in Title

### CODE REVIEW

File: src/commands/basemaps-mapsheet/create-mapsheet.ts [hunk 1]
```diff
@@ -77,6 +81,19 @@ export const basemapsCreateMapSheet = command({
...
 });
 
+/**
+ * Read Basemaps bundled config file by a given path can be both local or s3 path. Also support for gunzip json file.
+ *
+ */
+async function readConfig(config: string): Promise<ConfigBundled> {
+  const data = await fsa.read(config);
+  if (config.endsWith('.gz')) {
+    return JSON.parse((await pGunzip(data)).toString()) as ConfigBundled;
```
Comment (l0b0) [inline src/commands/basemaps-mapsheet/create-mapsheet.ts:91]: Is `promisify` + `await` a common pattern? I've not seen that one before. Would it be easier to do `return gunzip(data, (error, result) => JSON.parse(result))`?

**Review by l0b0 — COMMENTED** [review]
Could you please clarify the title and description? Based on the code it seems the title could be something like "feat(basemaps): Handle gzipped config file transparently".

File: src/commands/basemaps-mapsheet/create-mapsheet.ts (hunk 1 above)
Comment (Wentao-Kuang) [reply src/commands/basemaps-mapsheet/create-mapsheet.ts:91]: @blacha Adding this feature into [@chunkd](https://github.com/blacha/chunkd/pull/1345/files), I think we will close this and just upgrade to use the new @chunkd later.

Comment (blacha) [reply src/commands/basemaps-mapsheet/create-mapsheet.ts:91]: Javascript is a async language, so most things that have a `callback` don't actually return anything

```typescript 
const x = gunzip(data, (error, result) => JSON.parse(result))
console.log(typeof x, x) // "undefined", undefined
```

`promisfy` is a function that converts the old callback syntax into something you can return and use in the new `async/await` syntax

```typescript
const y = await pGunzip(data)
console.log(y) // data
```

@Wentao-Kuang We need this pull request still as upgrading to latest chunkd will be too much work for this story.

**Pull request closed without merging**